BasicGates/WorkBook_BasicGates1-4_VM3.ipynb

```diff
@@ -0,0 +1,983 @@
+    "    // Type X(q);\n",
+    "    // Then run the cell using Ctrl+Enter (⌘+Enter on macOS).\n",
+    "    X(q);\n",
+    "    // ...\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "disturbed-wilson",
+   "metadata": {},
+   "source": [
+    "[Return to Task 1 of the Basic Gates Quantum Kata Notebook  (opens in new tab)](./BasicGates.ipynb)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "satisfied-volunteer",
+   "metadata": {},
+   "source": [
+    "***"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "loved-korean",
+   "metadata": {},
+   "source": [
+    "## Task 1.2. Basis change: |0⟩ to |+⟩ and |1⟩ to |-⟩ (and vice versa)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "destroyed-scoop",
+   "metadata": {},
+   "source": [
+    "**Input:** A qubit in state |ψ⟩ = α |0⟩ + β |1⟩.\n",
+    "\n",
+    "**Goal:** Change the state of the qubit as follows:\n",
+    "If the qubit is in state |0⟩, change its state to |+⟩ = (|0⟩ + |1⟩) / sqrt(2).\n",
+    "If the qubit is in state |1⟩, change its state to |-⟩ = (|0⟩ - |1⟩) / sqrt(2).\n",
+    "If the qubit is in superposition, change its state according to the effect on basis vectors, as specified above."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "virtual-recall",
+   "metadata": {},
+   "source": [
+    "## Solution"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "twenty-proportion",
+   "metadata": {},
+   "source": [
+    "This is fundamentally book work, and a well-known property of the extremely important Hadamard (H) gate.  \n",
+    "It is easy to see how it works by examining its matrix representation.   \n",
+    "See [H](https://docs.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.intrinsic.h)  \n",
+    "The H gate performs the exact state change given in the question."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "turkish-starter",
+   "metadata": {},
+   "source": [
+    "The input superposition is given as $|\\psi\\rangle = 0.6|0\\rangle + 0.8|1\\rangle$"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "saving-biography",
+   "metadata": {},
+   "source": [
+    "Since 1/(sqrt(2) =0.7071 (to 4 decimal places) we have, for the goal, that:\n",
+    "\n",
+    "If the qubit is in state |0⟩, change its state to |+⟩ = 0.7071(|0⟩ + |1⟩)\n",
+    "\n",
+    "If the qubit is in state |1⟩, change its state to |-⟩ = 0.7071(|0⟩ - |1⟩)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "accessible-device",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "We know that H\n",
+    "$$\n",
+    "=\n",
+    "0.7071\\begin{bmatrix}\n",
+    "   1 & 1 \\\\\n",
+    "   1 & -1\n",
+    "  \\end{bmatrix}\n",
+    "$$"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "inner-basketball",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "And we have\n",
+    "$$\n",
+    "H(|ψ⟩) = \n",
```

Comment:
Formulas written as a mix of plain text and formulas are very hard to read, and so are formulas split across multiple lines if each line ends up very short. It's better to write the whole equation as a formula, and group multiple parts of the equation in one line, so that the whole formula can be read at a glance. You can see examples in other workbooks and tutorials